[xpu][static_launcher] 2/4: Refactor static cuda launcher to static triton launcher.

etaf opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #171450
* #171432
* #169938
* #168952
* __->__ #169121

This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

etaf committed:
[Inductor][refactor] rename StaticallyLaunchedCudaKernel to StaticallyLaunchedTritonKernel

[ghstack-poisoned]

etaf committed:
Update on "[Inductor][refactor] Rename static cuda launcher to static triton launcher."




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/3: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- static_cuda_launcher.py -> static_triton_launcher.py
- torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda/static_cuda_launcher.cpp


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/3: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/3: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/3: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/3: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/3: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/3: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/3: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/3: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/3: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

EikanWang commented on torch/csrc/inductor/static_launcher/cuda/static_cuda_launcher.cpp:
I'd prefer the latter one as the file names include the device information.

etaf committed:
Update on "[xpu][static_launcher] 2/3: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/4: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/4: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/4: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/4: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/4: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/4: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/4: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/static_cuda_launcher.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/4: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf committed:
Update on "[xpu][static_launcher] 2/4: Refactor static cuda launcher to static triton launcher."


This PR refactor the cuda static launcher to triton static launcher to be able to be reused by XPU triton backend as follow:
- Rename class `StaticallyLaunchedCudaKernel` -> class `StaticallyLaunchedTritonKernel`, and then keep `StaticallyLaunchedCudaKernel` inherit the  `StaticallyLaunchedTritonKernel`. Then there will be `StaticallyLaunchedXpuKernel`.
- Rename static_cuda_launcher.py -> static_triton_launcher.py
- Rename torch/csrc/inductor/static_cuda_launcher.cpp -> torch/csrc/inductor/static_launcher/cuda.cpp
- Rename test/_inductor/test_static_cuda_launcher.py -> test/_inductor/test_static_triton_launcher.py


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo

[ghstack-poisoned]

etaf commented:
@pytorchbot merge

etaf commented:
@pytorchbot drci
